extension/flat_tensor/serialize/serialize.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/extension/flat_tensor/serialize/serialize.h>
+#include <executorch/extension/flat_tensor/serialize/schema_generated.h>
+#include <executorch/extension/flat_tensor/serialize/scalar_type_generated.h>
+
+#include <string>
+#include <fstream>
+
+namespace executorch {
+namespace extension {
+namespace flat_tensor {
+
+namespace {
+  size_t padding_required(size_t offset, size_t alignment) {
+    // Returns the padding required to align `offset` to `alignment`.
+    size_t remainder = offset % alignment;
+    if (remainder != 0) {
+        return alignment - remainder;
+    }
+    return 0;
+  }
+
+  size_t aligned_size(size_t input_size, size_t alignment) {
+    // Returns input_size padded up to the next whole multiple of alignment.
+    return input_size + padding_required(input_size, alignment);
+  }
+
+  void serialize_nulls(std::ostream& out, size_t num_bytes) {
+    for (size_t i = 0; i < num_bytes; i++) {
+      out.write("\0", 1);
+    }
+  }
+}
+
+ET_EXPERIMENTAL runtime::Error save_ptd(
+  const std::string& path,
+  const std::map<std::string, exec_aten::Tensor>& tensor_map,
+  const size_t tensor_alignment
+) {
+  // Assert the system is little endian
```

Comment:
Not really necessary because the other save_ptd will do the same when this calls it